### PR TITLE
Add cloud spanner and mssql compact to db specifications

### LINKED ISSUE #3106: Proposed additional dbs for spec
**What are you trying to achieve?**
Including gcp spanner and mssql compact as part of db.system spec. 

Proposed naming `spanner` and `mssqlcompact`

**Additional context.**

Related to entity instrumentation that i'm working on, neither are currently in the spec.
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/898

### PR DESCRIPTION
**Pull request by mrblonde91**
Fixes #3106

## Changes
Added Microsoft Sql Server Compact and Cloud Spanner to db spec. 

For non-trivial changes, follow the [change proposal process](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CONTRIBUTING.md#proposing-a-change) and link to the related issue(s) and/or [OTEP(s)](https://github.com/open-telemetry/oteps), update the [`CHANGELOG.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md), and also be sure to update [`spec-compliance-matrix.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md) if necessary.

Related issues #

Related [OTEP(s)](https://github.com/open-telemetry/oteps) #

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ release.
 - Remove spaces from example exporter User-Agent header to conform to RFC7231 & RFC7230.
   [#3052](https://github.com/open-telemetry/opentelemetry-specification/pull/3052)
 
+- Add Cloud Spanner and Microsoft Sql to db.system semantic conventions.
```
Comment (Kielek) [inline CHANGELOG.md:13]: Microsoft Sql -> Microsoft SQL Server Compact

File: specification/trace/semantic_conventions/database.md [hunk 2]
```diff
@@ -117,6 +111,8 @@ Some database systems may allow a connection to switch to a different `db.user`,
 | `cockroachdb` | CockroachDB |
 | `opensearch` | OpenSearch |
 | `clickhouse` | ClickHouse |
+| `spanner` | Cloud Spanner |
+| `mssqlcompact` | Microsoft SQL Server |
```
Comment (Kielek) [inline specification/trace/semantic_conventions/database.md:115]: I suppose that you have manually modified this file. It should be generated based on database.yaml file (description is different that brief value for mssqcompact.

Comment (mrblonde91) [reply specification/trace/semantic_conventions/database.md]: Yep hadn't noticed that. Ran it now and it's fixed.

**Review by MovieStoreGuy — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

File: semantic_conventions/trace/database.yaml [hunk 3]
```diff
@@ -160,6 +160,12 @@ groups:
             - id: clickhouse
               value: 'clickhouse'
               brief: 'ClickHouse'
+            - id: spanner
+              value: 'spanner'
+              brief: 'Cloud Spanner'
+            - id: mssqlcompact
```
Comment (carlosalberto) [inline semantic_conventions/trace/database.yaml:166]: Can we put this next to the existing `mssql` entry? keeping them next to each other simplifies reading in general.

**Review by carlosalberto — APPROVED** [review]

File: semantic_conventions/trace/database.yaml (hunk 3 above)
Comment (mrblonde91) [reply semantic_conventions/trace/database.yaml]: Done

**Review by MrAlias — APPROVED** [review]

**Pull request merged**